### PR TITLE
Refactoring for janus.js release on npm

### PR DESCRIPTION
**Pull request by atoppi**
Since we have taken the ownership of the [janus-gateway package on npm](https://www.npmjs.com/package/janus-gateway), this PR refactors the project for the release of `janus.js` in the npm public registry.

**Potentially breaking changes**
- `package.json` in the project root **has been removed**
- js modules are now built under `npm/dist` 
- `./npm/package-lock.json` has been added (deps versions updated)
- `./npm/package.json` has been refactored

**How janus.js packages will be published**

We have added a `prerelease` script to `./npm/package.json` that generates an ES version of `janus.js` through `rollup` under `./npm/dist/janus.es.js` and copy `janus.js` under `./npm/src/janus.js`

So basically the publishing cycle will be the following (remove `--dry-run`)
```
# update version in ./npm/package.json
# npm run prerelease
# npm publish --dry-run
```

The package published on npm will have the following contents:
```
README.md        ---> the README that will be shown on npm
dist/janus.es.js ---> the "main" of the package.json as generated by rollup
janus.d.ts       ---> the "types" of the package.json
package.json    
src/janus.js     ---> copied from "html/janus.js" when doing "npm run prerelease"
```

Feedbacks are very welcome, in particular about the project structure we defined and how we think to manage the release cycle.

### CODE REVIEW

**Comment by lminiero** [comment]
Merging :+1:
